src/NUnitConsole/nunit3-console/TestEventHandler.cs

```diff
@@ -133,7 +137,44 @@ private void TestOutput(XmlNode outputNode)
         {
             var testName = outputNode.GetAttribute("testname");
             var stream = outputNode.GetAttribute("stream");
+            if (stream == "Error" && _outWriter != _errWriter)
+                TestErrorOutput(outputNode, testName);
+            else
+                TestOutput(outputNode, testName, stream);
+        }
+
+        private string _currentErrorLabel;
```

Comment:
Nitpick - could this field move to live with the other similar ones at the top of the class?